pkg/trace/api/tracer_flare.go

```diff
@@ -0,0 +1,74 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2023-present Datadog, Inc.
+
+package api
+
+import (
+	stdlog "log"
+	"net/http"
+	"net/http/httputil"
+	"net/url"
+	"regexp"
+	"strings"
+	"time"
+
+	"github.com/DataDog/datadog-agent/pkg/trace/log"
+)
+
+const (
+	serverlessFlareEndpointPath = "/api/ui/support/serverless/flare"
+)
+
+var ddURLRegexp = regexp.MustCompile(`^app(\.[a-z]{2}\d)?\.(datad(oghq|0g)\.(com|eu)|ddog-gov\.com)$`)
+var versionNumsddURLRegexp = regexp.MustCompile(`(\d+)\.(\d+)\.(\d+)`)
+
+type endpointGetter func(url *url.URL, agentVersion string)
+
+// tracerFlareTransport forwards request to tracer flare endpoint.
+type tracerFlareTransport struct {
+	rt           http.RoundTripper
+	getEndpoint  endpointGetter
+	agentVersion string
+}
+
+func getServerlessFlareEndpoint(url *url.URL, agentVersion string) {
+	// The DNS doesn't redirect to the proper endpoint when a subdomain is not present in the baseUrl.
+	// Adding app. subdomain here for site like datadoghq.com
+	if !ddURLRegexp.MatchString(url.Host) {
+		url.Host = "app." + url.Host
+	} else {
+		// Following exisiting logic to prefixes the domain with the agent version
+		// https://github.com/DataDog/datadog-agent/blob/e9056abe94e8dbddd51bbc901036e7362442f02e/pkg/config/utils/endpoints.go#L129
+		subdomain := strings.Split(url.Host, ".")[0]
+		versionNums := strings.Join(versionNumsddURLRegexp.FindStringSubmatch(agentVersion)[1:], "-")
+		newSubdomain := versionNums + "-flare"
+
+		url.Host = strings.Replace(url.Host, subdomain, newSubdomain, 1)
+	}
```

Comment:
Any way we could just use `AddAgentVersionToDomain` instead, so we don't have to copy the code, then do whatever other manipulations we need to do on top of that function?

Reply:
`AddAgentVersionToDomain` is from 
 github.com/DataDog/datadog-agent/pkg/config/util, I was concerned about pulling in an entire pkg for a couple of lines. What do you think?